src/bsoncxx/test/bson_get_values.cpp

```diff
@@ -0,0 +1,154 @@
+#include "catch.hpp"
+
+#include <bsoncxx/builder/stream/document.hpp>
+#include <bsoncxx/builder/stream/helpers.hpp>
+
+using namespace bsoncxx;
+
+TEST_CASE("[] can reach into nested arrays", "[bsoncxx]") {
+    using namespace builder::stream;
+
+    builder::stream::document build_doc;
+    // {
+    //     "ints": [ 1, 3, [ 5 ] ],
+    //     "bools": [ true, false ]
+    // }
+    build_doc << "ints" << open_array << 1 << 3 << open_array << 5 << close_array << close_array
+              << "bools" << open_array << true << false << close_array;
+
+    auto doc = build_doc.view();
+
+    SECTION("works with one level") {
+        REQUIRE(doc["ints"][0]);
+        REQUIRE(doc["ints"][0].get_int32() == 1);
+
+        REQUIRE(doc["ints"][1]);
+        REQUIRE(doc["ints"][1].get_int32() == 3);
```

Comment:
it would be so nice if operator== could work without the get_int32() method, maybe some day